src/memory/buffer.cc

```diff
@@ -19,34 +19,34 @@
 
 namespace swoole {
 
+BufferChunk::BufferChunk(Type type, uint32_t size) : type(type), size(size) {
+    if (type == TYPE_DATA && size > 0) {
```

Comment:
I think the if condition of the type should be removed, and if the size is greater than 0, the memory will be allocated

Reply:
Yes, there is an intrinsic relationship between these two parameters according to the original design.

If the "type" conditional judgment was removed, all use cases which `type` is not `TYPE_DATA` must insure `size` equal to zero since `value` property is union type.

```cpp
BufferChunk *chunk = out_buffer->alloc(BufferChunk::TYPE_SENDFILE, 0);
chunk->value.object = task.release();
chunk->destroy = Socket_sendfile_destructor;
```

If the "type" conditional judgment was preserved, `size` would be ignored in these cases.

So I made conservative changes.